app/app.go

```diff
@@ -331,21 +339,51 @@ func NewOsmosisApp(
 		if !ok {
 			panic(fmt.Sprintf("failed to retrieve %s from config.toml", rpcAddressConfigName))
 		}
-		nodeStatusChecker := service.NewNodeStatusChecker(rpcAddress)
+		nodeStatusChecker := sqsservice.NewNodeStatusChecker(rpcAddress)
 
 		// Create the SQS streaming service by setting up the write listeners,
 		// the SQS ingester, and the pool tracker.
-		sqsStreamingService := service.New(writeListeners, storeKeyMap, sqsIngester, poolTracker, nodeStatusChecker)
+		sqsStreamingService := sqsservice.New(writeListeners, storeKeyMap, sqsIngester, poolTracker, nodeStatusChecker)
+
+		streamingServices = append(streamingServices, sqsStreamingService)
+	}
+
+	// Initialize the config object for the indexer
+	indexerConfig := indexer.NewConfigFromOptions(appOpts)
+
+	// initialize indexer if enabled
+	if indexerConfig.IsEnabled {
+		indexerIngester := indexerConfig.Initialize()
+
+		// TODO: handle graceful shutdown
+		pubSubCtx := context.Background()
```

Comment:
Note: let's keep this in a task to come back to

Reply:
https://github.com/osmosis-labs/osmosis/pull/8411#discussion_r1649071670

Reply:
[Linear issue created and assigned](https://linear.app/osmosis/issue/IND-50/handle-graceful-shutdown-for-pubsubctx)